internal/pkg/agent/application/pipeline/dispatcher/dispatcher.go

```diff
@@ -74,7 +76,25 @@ func (ad *ActionDispatcher) key(a fleetapi.Action) string {
 }
 
 // Dispatch dispatches an action using pre-registered set of handlers.
-func (ad *ActionDispatcher) Dispatch(acker store.FleetAcker, actions ...fleetapi.Action) error {
+// ctx is used here ONLY to carry the span, for cancelation use the cancel
+// function of the ActionDispatcher.ctx.
+func (ad *ActionDispatcher) Dispatch(ctx context.Context, acker store.FleetAcker, actions ...fleetapi.Action) (err error) {
+	span, ctx := apm.StartSpan(ctx, "dispatch", "app.internal")
+	defer func() {
+		apm.CaptureError(ctx, err).Send()
+		span.End()
+	}()
+	// Creating a child context that carries both the ad.ctx cancelation and
+	// the span from ctx.
+	ctx, cancel := context.WithCancel(ad.ctx)
+	// golint disallows us from discarding the cancel func, even
+	// though we know that the resources for ctx will be released
+	// when ad.ctx is canceled.
+	// ctx is no longer needed after calling acker.Commit(ctx), so we can
+	// safely call cancel().
+	defer cancel()
+	ctx = apm.ContextWithSpan(ctx, span)
```

Comment:
Agree, on the comment here, let's add a `nolint` rule here, https://golangci-lint.run/usage/false-positives/